libs/cvr-fixture-generator/src/cli/generate/main.ts

```diff
@@ -219,72 +220,84 @@ export async function main(
+          ballotType: BallotType.Precinct,
+          ballotHash,
+          isTestMode: testMode,
+          precinctId: castVoteRecord.BallotStyleUnitId,
+        });
+        for (const i of [0, 1] as const) {
+          const imageFilePath = join(
+            castVoteRecordDirectory,
+            assertDefined(castVoteRecord.BallotImage[i]?.Location).replace(
+              'file:',
+              ''
+            )
+          );
+          const layoutFilePath = imageFilePath.replace('.jpg', '.layout.json');
+          const existingImagePath = imagePathsByPaperSize.get(
+            election.ballotLayout.paperSize
+          );
 
-        const { width, height } = ballotPaperDimensions(
-          election.ballotLayout.paperSize
-        );
-        const pageDpi = 200;
-        await writeImageData(
-          imageFilePath,
-          createImageData(
-            new Uint8ClampedArray(width * pageDpi * height * pageDpi * 4),
-            width * pageDpi,
-            height * pageDpi
-          )
-        );
+          if (existingImagePath) {
+            await fs.copyFile(existingImagePath, imageFilePath);
+          } else {
+            const { width, height } = ballotPaperDimensions(
+              election.ballotLayout.paperSize
+            );
+            const pageDpi = 200;
 
-        const layout = layouts[i];
-        let layoutFileHash = sha256('bmd-ballot');
-        if (layout) {
-          const layoutFileContents = JSON.stringify(layout);
-          fs.writeFileSync(layoutFilePath, layoutFileContents);
-          layoutFileHash = sha256(layoutFileContents);
-        }
+            const imageData = createImageData(
+              new Uint8ClampedArray(width * pageDpi * height * pageDpi * 4),
+              width * pageDpi,
+              height * pageDpi
+            );
+            await writeImageData(imageFilePath, imageData);
+            imagePathsByPaperSize.set(
+              election.ballotLayout.paperSize,
+              imageFilePath
+            );
+          }
 
-        const imageHash = sha256(fs.readFileSync(imageFilePath));
-        populateImageAndLayoutFileHashes(
-          assertDefined(castVoteRecord.BallotImage[i]),
-          { imageHash, layoutFileHash }
-        );
-      }
-    }
+          const layout = layouts[i];
+          let layoutFileHash = sha256('bmd-ballot');
+          if (layout) {
+            const layoutFileContents = JSON.stringify(layout);
+            await fs.writeFile(layoutFilePath, layoutFileContents);
+            layoutFileHash = sha256(layoutFileContents);
+          }
 
-    const castVoteRecordReport: CVR.CastVoteRecordReport = {
-      ...reportMetadata,
-      CVR: [castVoteRecord],
-    };
-    fs.writeFileSync(
-      join(
-        castVoteRecordDirectory,
-        CastVoteRecordExportFileName.CAST_VOTE_RECORD_REPORT
-      ),
-      JSON.stringify(castVoteRecordReport)
-    );
-  }
+          const imageHash = sha256(await fs.readFile(imageFilePath));
```

Comment:
Should we save the hash as well, if it's the same file? Maybe this part is fast enough that it doesn't matter

Reply:
Added. It's a small difference, but might as well.